app.R

```diff
@@ -1,1104 +1,348 @@
+    }
+    else if(input$type == "Protein vs mRNAs") {
+      updateSelectizeInput(session, "genes", label = "Enter Gene/Protein to plot", choices = genenames_corr$V1, server = TRUE)
+      shinyjs::show(id = "genes")
+      shinyjs::hide(id = "gene")
+      shinyjs::hide(id = "subtype")
+      subtype <- NULL
+    }
+    updateSelectizeInput(session, 'subtype', choices = subtype, server = TRUE)
+    if(input$type != "Protein vs mRNAs") {shinyjs::show(id = "subtype")}
+  })
+  
+  # Handle event when user selects subtype
+  observeEvent(input$subtype, {
+    # Phosphosite doesn't allow you to toggle clinical options
+    if(input$type == "Phosphosite") {
+      shinyjs::hide(id = "subtype_options")
+    }
+    else {
+      shinyjs::show(id = "subtype_options")
+    }
+    shinyjs::show(id = "gene")
+    shinyjs::hide(id = "genes")
+    shinyjs::hide(id = "mutation_status")
+    
+    if(input$subtype == "Multiplot") {
+      shinyjs::show(id = "genes")
+      shinyjs::hide(id = "subtype_options")
+      shinyjs::hide(id = "gene")
+      updateSelectizeInput(session, "genes", label = paste("Multi",tolower(itemPlot)," plot",sep=""))
+    }
+    else if(input$subtype == "Subtype") {
+      updateSelectizeInput(session, "gene", label = paste(itemPlot,"to plot"))
+      query <- paste0("SELECT DISTINCT FAB FROM ",input$type,"_Clinical;")
+      subtype_choices <- unlist(dbGetQuery(database,query),use.names = FALSE)
+      subtype_choices <- str_sort(subtype_choices)
+      updateCheckboxGroupInput(session,
+                               "subtype_options",
+                               label = "Subtypes",
+                               choices = subtype_choices,
+                               selected = subtype_choices)
+    }
+    else if(input$subtype == "Cytogenetics") {
+      updateSelectizeInput(session, "gene", label = paste(itemPlot,"to plot"))
+      query <- paste0("SELECT DISTINCT Cyto_Risk FROM ",input$type,"_Clinical;")
+      cyto_choices <- unlist(dbGetQuery(database,query),use.names = FALSE)
+      cyto_choices <- str_sort(cyto_choices)
+      updateCheckboxGroupInput(session,
+                               "subtype_options",
+                               label = "Criteria",
+                               choices = cyto_choices,
+                               selected = cyto_choices)
+    }
+    else if(input$subtype == "Fusion") {
+      updateSelectizeInput(session, "gene", label = paste(itemPlot,"to plot"))
+      query <- paste0("SELECT DISTINCT Fusion FROM ",input$type,"_Clinical;")
+      fusion_choices <- unlist(dbGetQuery(database,query),use.names = FALSE)
+      fusion_choices <- str_sort(fusion_choices)
+      updateCheckboxGroupInput(session,
+                               "subtype_options",
+                               label = "Criteria",
+                               choices = fusion_choices,
+                               selected = fusion_choices)
+    }
```

Comment:
Seems like these repetitive blocks could be candidates for future collapsing into a function (especially if the number of variables increases). totally fine right now